src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -2195,17 +2195,26 @@ case class TableExplode(child: TableIR, column: String) extends TableIR {
 
     val childRowType = child.typ.rowType
 
+    val fieldToExplode = columnType match {
+      case TArray(_, _) =>
+        GetField(Ref("row", childRowType), column)
+      case TSet(_, _) =>
+        ToArray(GetField(Ref("row", childRowType), column))
+      case _ =>
+        fatal("expected field to explode to be an array or set")
```

Comment:
you could give a bit more info: `s"expected field to explode to be an array or set, found $fieldType"`